opentasks/src/main/java/org/dmfs/tasks/homescreen/TaskListWidgetUpdaterService.java

```diff
@@ -233,7 +233,7 @@ public RemoteViews getViewAt(int position)
                 row.setTextViewText(android.R.id.text1, mDueDateFormatter.format(dueDate, DateFormatContext.WIDGET_VIEW));
 
                 // highlight overdue dates & times
-                if ((!dueDate.allDay && dueDate.before(mNow) || dueDate.allDay
+                if ((!dueDate.allDay && Time.compare(dueDate, mNow) <= 0 || dueDate.allDay
```

Comment:
This looks unrelated to the issue. Why did you make this change?

Reply:
While working on this fix I observed some cases where the update was processed right within the second of the due date. I noticed that in those cases the task did not turn red but remained white in the widget. That seemed wrong. The reason was that dueDate was not before but exactly equal to mNow – at least in `android.text.format.Time`'s second precision.

Therefore I changed this comparison essentially from `dueDate < mNow` to `dueDate <= mNow`. Sorry, if the commit message was not clear about my intention.